prog.go

```diff
@@ -308,6 +308,45 @@ func (p *BPFProg) AttachXDP(deviceName string) (*BPFLink, error) {
 	return bpfLink, nil
 }
 
+func (p *BPFProg) AttachXDPLegacy(deviceName string, flag XDPFlags) error {
+	optsC := &C.struct_bpf_xdp_attach_opts{
+		sz:          C.ulong(unsafe.Sizeof(C.struct_bpf_xdp_attach_opts{})),
+		old_prog_fd: 0,
+	}
```

Comment:
Could you convert these instantiations to use the `structure handlers` pattern found in `libbpfgo.h`, since cgo has problems with structure packaging and field offsets? By instance:

- cgo_bpf_xdp_attach_opts_new
- cgo_bpf_xdp_attach_opts_free

Reply:
done